src/bokeh/plotting/_tools.py

```diff
@@ -170,15 +170,15 @@ def _resolve_tools(tools: str | Sequence[Tool | str]) -> tuple[list[Tool], dict[
     tool_map: dict[str, Tool] = {}
 
     if not isinstance(tools, str):
-        temp_tool_str = ""
+        temp_tool_str = []
         for tool in tools:
             if isinstance(tool, Tool):
                 tool_objs.append(tool)
             elif isinstance(tool, str):
-                temp_tool_str += tool + ','
+                temp_tool_str.append(tool)
             else:
                 raise ValueError("tool should be a string or an instance of Tool class")
-        tools = temp_tool_str
+        tools = ",".join(temp_tool_str)
```

Comment:
This change has neglectable benefits. It is true that joining a list of string is faster but in general the input is already a string. If not, the saved time is in the range of ns. This is acceptable.

Please revert this change, to keep the change as minimal as possible.